### DISCUSSION TITLE
create connection with oracle wallet

### DISCUSSION THREAD

**Discussion by hphan10** [oracle/python-oracledb#233] [Q&A] [answered]
Hello,

on cx_Oracle I was able to use these

```python
os.environ['TNS_ADMIN'] = 'sqlnet.ora  and tnsnames.ora location'
connection = cx_Oracle.connect(dsn='DSN NAME')
```
but on oracledb it gave me error
```python
os.environ['TNS_ADMIN'] = 'sqlnet.ora  and tnsnames.ora location'
connection = oracledb.connect(dsn='DSN NAME')
```
`oracledb.exceptions.DatabaseError: DPY-4001: no credentials specified`

**Comment by anthony-tuininga** [comment, marked as answer]
Thin mode (the default) does not support external authentication which is what you are showing here. You need to add this line before calling `oracledb.connect(dsn="DSN NAME")`:

```python
oracledb.init_oracle_client()
```

This enables thick mode which is what cx_Oracle used. At that point your program should work the same way it did before!

**Reply by cjbj** [reply]
The doc on using Thick mode is at https://python-oracledb.readthedocs.io/en/latest/user_guide/initialization.html

**Reply by hphan10** [reply]
Thank you so much @anthony-tuininga.
